### ISSUE TITLE
[saml] Azure AD freaks out when AuthnRequest's ID starts with a digit

### ISSUE THREAD

**Issue by srenatus** [coreos/dex#986]
I've found that the following AuthnRequest generated by dex' SAML connector let's me login successfully:

```xml
<AuthnRequest 
    xmlns="urn:oasis:names:tc:SAML:2.0:protocol" 
    ID="qewzzsakl7npvinrooe3cpt5uq" 
    Version="2.0" 
    IssueInstant="2017-07-07T08:20:27Z" 
    Destination="https://login.windows.net/MY_APP_ID/saml2" 
    ProtocolBinding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST" 
    AssertionConsumerServiceURL="http://127.0.0.1:5556/dex/callback">
...
```

while this one,

```xml
<AuthnRequest 
    xmlns="urn:oasis:names:tc:SAML:2.0:protocol" 
    ID="6dyptp6enwgwwvuz4pkfiydly4" 
    Version="2.0" 
    IssueInstant="2017-07-07T08:21:53Z" 
    Destination="https://login.windows.net/MY_APP_ID/saml2" 
    ProtocolBinding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST" 
    AssertionConsumerServiceURL="http://127.0.0.1:5556/dex/callback">
...
```

leads to the following error screen:

<img width="338" alt="screen shot 2017-07-07 at 10 26 21" src="https://user-images.githubusercontent.com/870638/27949667-c952c77c-62fe-11e7-8a0d-2280cfcb106a.png">

This [random, seemingly related Stackoverflow post](https://stackoverflow.com/a/29798747) says
> it must merely conform to the standards of an XML ID. An XML ID is an xsd:NCName, which is derived from xsd:Name, *which can't start with a number or contain spaces*, and should have 160 bits of "randomness".

**Comment by kkohtaka** [comment]
I encountered the same issue and the following PR will fix this issue.

#985

**Comment by srenatus** [comment]
Fixed by #985. Thanks a lot @kkohtaka 🎉 😃
